### DISCUSSION TITLE
GPL-3.0 License

### DISCUSSION THREAD

**Discussion by bjlittle** [pyvista/scikit-gmsh#385] [Q&A] [answered]
@tkoyama010 Is there a specific reason why you've chosen to distribute `scikit-gmsh` under a GPL-3.0 license?

This might not be attractive to some users who would prefer a more permissive option such as [Apache-2.0](https://spdx.org/licenses/Apache-2.0.html), [BSD-3-Clause](https://spdx.org/licenses/BSD-3-Clause.html) or [MIT](https://spdx.org/licenses/MIT.html).

Just curious, thanks 😄

**Comment by tkoyama010** [comment, marked as answer]
Good question. It is because Gmsh is  GPL-3.0 license. 
> Linking a GPL covered work statically or dynamically with other modules is making a combined work based on the GPL covered work. Thus, the terms and conditions of the GNU General Public License cover the whole combination. See also [What legal issues come up if I use GPL-incompatible libraries with GPL software?](https://www.gnu.org/licenses/gpl-faq.en.html#GPLIncompatibleLibs)

https://www.gnu.org/licenses/gpl-faq.en.html#GPLStaticVsDynamic

This is also the reason why this library is separated from the rest of the PyVista ecosystem. It seems that [pygmsh](https://github.com/nschloe/pygmsh) also follows this and is licensed under the GPL.
